src/fga/utils/interface-check.ts

```diff
@@ -0,0 +1,17 @@
+import { Subject, ResourceInterface } from '../interfaces';
+
+export function isSubject(resource: any): resource is Subject {
+  return (
+    Object.prototype.hasOwnProperty.call(resource, 'resourceType') &&
+    Object.prototype.hasOwnProperty.call(resource, 'resourceId')
+  );
+}
+
+export function isResourceInterface(
+  resource: any,
+): resource is ResourceInterface {
+  return (
+    resource.getResourceType !== undefined &&
+    resource.getResourceId !== undefined
+  );
+}
```

Comment:
We typically prefer `unknown` over `any`. In this case, the `any` would allow `null` or `undefined` bass which would cause this function to throw.

How about this?

```suggestion
export function isResourceInterface(
  resource: unknown,
): resource is ResourceInterface {
  return (
    !!resource &&
    typeof resource === 'object' &&
    'getResourceType' in resource &&
    'getResourceId' in resource
  );
}
```